---
title: "Duplicate Lymphoma Survival Analysis"
output: html_document
date: "`r Sys.Date()`"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Survival Analysis for Mature B-Cell Lymphomas (TCGA-DLBC)

```{r}
# load my libraries
library(TCGAbiolinks)
library(survminer)
library(survival)
library(SummarizedExperiment)
library(tidyverse)
library(DESeq2)
library(pander)
```

```{r}
clin_dlbc <- GDCquery_clinic('TCGA-DLBC')
which(colnames(clin_dlbc) %in% c('vital_status', 'days_to_last_follow_up', 'days_to_death'))
clin_dlbc[,c(9,37,43)]
```

```{r}
table(clin_dlbc$vital_status)
```

```{r}
clin_dlbc$deceased <- ifelse(clin_dlbc$vital_status == 'Alive', FALSE, TRUE)

```

```{r}
clin_dlbc$overall_survival <- ifelse(clin_dlbc$vital_status == 'Alive', clin_dlbc$days_to_last_follow_up, clin_dlbc$days_to_death)
```

```{r}
query_dlbc <- GDCquery(
  project = 'TCGA-DLBC',
  data.category = 'Transcriptome Profiling',
  experimental.strategy = 'RNA-Seq',
  workflow.type = 'STAR - Counts',
  sample.type = 'Primary Tumor',
  data.type = 'Gene Expression Quantification',
  access = 'open'
  )

output_dlbc <- getResults(query_dlbc)

```

```{r}
#tumor_ov <- output_ov[output_ov$sample_type == 'Primary Tumor', 'cases'][1:50]
#tumor_ov
```

```{r}
query_dlbc <- GDCquery(
  project = 'TCGA-DLBC',
  data.category = 'Transcriptome Profiling',
  experimental.strategy = 'RNA-Seq',
  workflow.type = 'STAR - Counts',
  sample.type = 'Primary Tumor',
  data.type = 'Gene Expression Quantification',
  access = 'open'
  #barcode = tumor
  )
```

```{r}
# download my data
GDCdownload(query_dlbc)
```

```{r}
# get counts
tcga_dlbc_data <- GDCprepare(query_dlbc, summarizedExperiment = TRUE)
dlbc_matrix <- assay(tcga_dlbc_data, 'unstranded')
dlbc_matrix[1:10, 1:10]
```

```{r}
# extract raw data and store in matrix
gene_metadata_dlbc <- as.data.frame(rowData(tcga_dlbc_data))
coldata_dlbc <- as.data.frame(colData(tcga_dlbc_data))
```

```{r}
dds <- DESeqDataSetFromMatrix(countData = dlbc_matrix,
                              colData = coldata_dlbc,
                              design = ~ 1)

keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep,]

vsd <- vst(dds, blind=FALSE)
dlbc_matrix_vst <- assay(vsd)
dlbc_matrix_vst[1:10, 1:10]
# vst means variance stabilized transformed counts
```

```{r}
# Get data for CDK11B only
dlbc_cdk11 <- dlbc_matrix_vst %>%
  as.data.frame() %>%
  rownames_to_column(var = 'gene_id') %>%
  gather(key = 'case_id', value = 'counts', -gene_id) %>%
  left_join(., gene_metadata_dlbc, by = 'gene_id') %>%
  filter(gene_name == 'CDK11B')
  
```

```{r}
# divide patients into low expression and high expression groups - I usually use the median value but you can use percentiles as well. In this case I have used percentiles to see if that changes the look of km survival curve. 

max_value <- quantile(dlbc_cdk11$counts, 0.7)
min_value <- quantile(dlbc_cdk11$counts, 0.3)

dlbc_cdk11$strata <- ifelse(dlbc_cdk11$counts >= max_value, 'HIGH', 'LOW')
```

```{r}
# Add all the information into one data frame
# remove additional characters in the case_id so that it matches with the submitter_id in the other data sets

dlbc_cdk11$case_id <- gsub('-01.*', '', dlbc_cdk11$case_id)
dlbc_cdk11 <- merge(dlbc_cdk11, clin_dlbc, by.x = 'case_id', by.y = 'submitter_id')
```

```{r}
# Survival Curve
fit <- survfit(Surv(overall_survival, deceased) ~ strata, data = dlbc_cdk11)
pander(fit)
```

```{r}
# Plot the survival curve
ggsurvplot(fit,
           data = dlbc_cdk11,
           pval = T,
           risk.table = T)


```

## Log Rank Test
That p-value of 0.57 indicated that there is no statistically significant difference. between the 2 groups of patients. This is contrary to what has been published about CDK11 and ovarian cancer. We are supposed to see a difference in survival between the 2 groups. I assume there are 2 factors affecting these results. One of them is that CDK11 is significantly expressed more in recurrent tumors or metastatic tumors and we only have 8 patient records of recurrent tumors. Running an analysis on

```{r}
# Use the survdiff function to compare between the 2 groups

fit2 <- survdiff(Surv(overall_survival, deceased) ~ strata, data = dlbc_cdk11)
pander(fit2)
```

